docs(AzureRedisInstance): user documentation (#449)
kyma-project/cloud-manager: docs/user/resources/04-40-10-azure-redis-instance.md

# AzureRedisInstance Custom Resource

The `azureredisinstance.cloud-resources.kyma-project.io` custom resource (CR) defines an
[Azure Cache for Redis instance](https://azure.microsoft.com/en-us/products/cache) that can be used as a caching solution within the cluster. 

> Please note that the current implementation does not support backups, so using `AzureRedisInstance` may not be ideal 
> for permanent storage. The endpoint and secret token for the instance are provided
> in a Secret resource, which is generated within the Kyma cluster. This Secret will share the same name as the `AzureRedisInstance`.

Reviewer: ```suggestion
> [!NOTE] 
> The current implementation does not support backups, so using `AzureRedisInstance` is not ideal for permanent storage. 
> The endpoint and secret token for the instance are provided in a Secret resource, which is generated within the Kyma cluster. This Secret shares the same name as the `AzureRedisInstance`.
```

> The Azure Cache for Redis `geo-location` will be set to match the location of the Kyma cluster where the manifest is applied.

Reviewer: ```suggestion
The Azure Cache for Redis `geo-location` is set to match the location of the Kyma cluster where the manifest is applied.
```

The underlying Redis tier is Premium, as it supports VirtualNetwork.

## Specification <!-- {docsify-ignore} -->

> This table lists the parameters of the given resource together with their descriptions. Note that this table is subject
> to change, as the underlying Azure API could change with the new versions. Therefore, use it is basic info and for any
> in-depth spec, please refer to [MS Azure latest Redis documentation](https://learn.microsoft.com/en-us/rest/api/redis/redis/create). 

Reviewer: ```suggestion
This table lists the parameters of the given resource together with their descriptions. The table is subject
to change, as the underlying Azure API can change with the new versions. Use it as basic information, and for
in-depth specification, see the [MS Azure latest Redis documentation](https://learn.microsoft.com/en-us/rest/api/redis/redis/create). 
```

> Also note that not all the parameters supported on the Azure side can be used from Kyma 
> (eg `properties.redisConfiguration.aof-backup-enabled`).

Reviewer: ```suggestion
> [!NOTE]
> Some parameters, such as **properties.redisConfiguration.aof-backup-enabled**, are supported by Azure, but you can't use them from Kyma.
```

**Spec:**

> 

Reviewer: Please consider adding another column for the required/optional value.

| **replicasPerPrimary**                                 | quantity | The number of replicas to be created per primary. Optional.                                       |                              
| **sku**                                                | object   | The SKU of the Redis cache to deploy. Required.                                                   |
| **sku.capacity**                                       | quantity | The size of the Redis cache to deploy. Valid values: 1, 2, 3, 4. Required.                        |
| **redisConfiguration**                                 | object   | Redis Settings. Not all settings supported on the Azure side are supported in Kyma. Optional.     |
| **redisConfiguration.maxclients**                      | string   | The max clients config. Optional.                                                                 |
| **redisConfiguration.maxfragmentationmemory-reserved** | string   | Value in megabytes reserved for fragmentation per shard. Optional.                                |

> | **redisConfiguration.maxmemory-delta**                 | string   | Value in megabytes reserved for non-cache usage per shard e.g. failover. Optional.                |

Reviewer: ```suggestion
| **redisConfiguration.maxmemory-delta**                 | string   | Value in megabytes reserved for non-cache usage per shard, for example, failover. Optional.                |
```

> | **redisConfiguration.maxmemory-policy**                | string   | The eviction strategy used when your data won't fit within its memory limit. Optional.            |

Reviewer: ```suggestion
| **redisConfiguration.maxmemory-policy**                | string   | The eviction strategy used when your data doesn't fit within its memory limit. Optional.            |
```

> | **redisConfiguration.maxmemory-reserved**              | string   | Value in megabytes reserved for non-cache usage per shard e.g. failover. Optional.                |

Reviewer: ```suggestion
| **redisConfiguration.maxmemory-reserved**              | string   | Value in megabytes reserved for non-cache usage per shard, for example, failover. Optional.                |
```

| **redisConfiguration.notify-keyspace-events**          | string   | The keyspace events which should be monitored. Optional.                                          |
| **redisConfiguration.zonal-configuration**             | string   | Zonal Configuration. Optional.                                                                    |

> An exemplary `AzureRedisInstance` custom resource:

Reviewer: ```suggestion
See an example of the `AzureRedisInstance` custom resource:
```
